### PR TITLE
Collatz Conjecture: test for non-integers

### PR DESCRIPTION
**Pull request by javidlakha**
The Collatz Conjecture holds only for positive integers. TypeScript does not have an integer type and therefore the input must be explicitly tested. 

Changes:
- Adds a test case for a positive non-integer
- Modifies the expected error message from 'Only positive numbers are allowed' to 'Only positive integers are allowed' so that the student can test the input using a single if statement

### CODE REVIEW

File: exercises/practice/collatz-conjecture/collatz-conjecture.test.ts [hunk 1]
```diff
@@ -22,16 +22,23 @@ describe('CollatzConjecture', () => {
...
 
   xit('negative value is an error', () => {
-    const expected = 'Only positive numbers are allowed'
+    const expected = 'Only positive integers are allowed'
     expect(() => {
       steps(-15)
     }).toThrowError(expected)
   })
+
+  xit('negative value is an error', () => {
```
Comment (SleeplessByte) [inline exercises/practice/collatz-conjecture/collatz-conjecture.test.ts:38]: This is not a negative value that's being tested.

**Comment by SleeplessByte** [comment]
I think this is a good change. Can you also propose this change in https://github.com/exercism/problem-specifications first? I expected other tracks benefitting from this addition as well.

File: exercises/practice/collatz-conjecture/collatz-conjecture.test.ts (hunk 1 above)
Comment (javidlakha) [reply exercises/practice/collatz-conjecture/collatz-conjecture.test.ts]: Corrected, thanks!

**Comment by javidlakha** [comment]
Yes, I shall do.

**Comment by SleeplessByte** [comment]
It took a _very_ long time, but as soon as @ErikSchierboom is back from xmas holiday, the problem-spec PR will be merged, which means we can _finally_ merge this as well. Thanks again!

**Review by SleeplessByte — APPROVED** [review]

**Pull request merged**